doc/src/modules/tensor/indexed.rst

```diff
@@ -2,5 +2,88 @@
 Indexed Objects
 ===============
 
-.. automodule:: sympy.tensor.indexed
+The classes ``IndexedBase``, ``Indexed``, and ``Idx`` represent a
+matrix element ``M[i, j]`` as in the following diagram::
+
+   1) The Indexed class represents the entire indexed object.
+              |
+           ___|___
+          '       '
+           M[i, j]
+          /   \__\______
+          |             |
+          |             |
+          |     2) The Idx class represents indices; each Idx can
+          |        optionally contain information about its range.
+          |
+    3) IndexedBase represents the 'stem' of an Indexed object, here M.
+
+There can be any number of indices on an ``Indexed`` object.  No symmetry
+properties are implemented in these objects. Implicit contraction of
+repeated indices is supported via the ``EinsteinSum`` function.
+
+Note that the support for complicated (i.e. non-atomic) integer
+expressions as indices is limited.  (This should be improved in
+future releases.)
+
+Basic Examples
+==============
+
+To express the above matrix element example:
+
+>>> from sympy import symbols, IndexedBase, Idx
+>>> M = IndexedBase('M')
+>>> i, j = symbols('i j', cls=Idx)
+>>> M[i, j]
+M[i, j]
+
+Repeated indices in a monomial imply summation when enclosed by the
+``EinsteinSum`` function. To express a matrix-vector product in terms of
+``Indexed`` objects:
+
+>>> from sympy import EinsteinSum
+>>> x = IndexedBase('x')
+>>> EinsteinSum(M[i, j] * x[j])
+EinsteinSum(x[j]*M[i, j])
+
+``EinsteinSum`` expressions can be differentiated:
+
+>>> EinsteinSum(x[j] * x[j]).diff(x[i])
+EinsteinSum(2*x[i])
+
+They can also be converted into Python functions accepting NumPy arrays:
+
+>>> func = EinsteinSum(M[i, j] * x[j]).numpify([M, x], [i])  # doctest: +SKIP
```

Comment:
In general, a Printer or subclass of Printer needs to have a method that shows how EinsteinSums are printed. Likely your numpify code needs to be moved to one of the base printers. Or you can add methods to EinstenSum that allow it to print itself, but the former is preferred.

Can you explain why the extra argument is required? Usually objects contain all the information they need to be able to print them selves.